openmc/deplete/operator.py

```diff
@@ -251,6 +273,40 @@ def _set_number_from_mat(self, mat):
             number = density * 1.0e24
             self.number.set_atom_density(mat_id, nuclide, number)
 
+    def _set_number_from_results(self, mat, prev_res):
+        """Extracts material nuclides and number densities.
+
+        If the nuclide concentration's evolution is tracked, the densities come
+        from depletion results.
+        Else, densities are extracted from the geometry in the summary.
+
+        Parameters
+        ----------
+        mat : openmc.Material
+            The material to read from
+
+        """
+        mat_id = str(mat.id)
+
+        # Get nuclide lists from geometry and depletion results
+        depl_nuc = prev_res[-1].nuc_to_ind.keys()
+        geom_nuc_densities = mat.get_nuclide_atom_densities()
+        geom_nuc = [x[0] for x in list(geom_nuc_densities.values())]
+
+        # Merge lists of nuclides
+        nuc_set = list(depl_nuc) + [n for n in geom_nuc if n not in depl_nuc]
```

Comment:
Python has a built-in `set` object for stuff like this.  You should be able to replace this with something like,
```python
nuc_set = set(depl_nuc) | set(geom_nuc)
```